frontend/src/components/ViewTrips/trip-utils.js

```diff
@@ -31,52 +36,109 @@ export function getCleanedTextInput(rawInput, defaultValue) {
+ * provided in the add trip form (all but the trip creator).
  *
  * TODO(#72 & #67): Remove 'remove empty fields' once there is better way to
  * remove collaborators (#72) and there is email validation (#67).
  *
  * @param {!string[]} collaboratorEmailsArr Array of emails corresponding
  *     to the  collaborators of the trip (not including the trip creator email).
  * @return {Promise<!string[]>} Promise that resolves to an array of the
- *     collaborator uids of the current user and the uids corresponding to the
- *     emails in `collaboratorEmailsArr.
+ *     collaborator uids corrsponding to the emails in `collaboratorEmailsArr`.
  */
 export async function getCollaboratorUidArray(collaboratorEmailArr) {
-  collaboratorEmailArr = [authUtils.getCurUserEmail()]
-                             .concat(collaboratorEmailArr);
-
-  // Removes empty fields (temporary until fix #67 & #72).
   const cleanedCollaboratorEmailArr = collaboratorEmailArr.filter(email => {
     return email !== '';
   });
 
   return await authUtils.getUserUidArrFromUserEmailArr(cleanedCollaboratorEmailArr);
 }
 
+/**
+ * Returns the new `accepted_collaborator_uid_arr` field value for the updated
+ * trip document.
+ *
+ * @param {!string[]} formCollabUidArr Array of the collaborator uids
+ *     corresponding to the emails in the `collaborator_email_arr` field of the
+ *     {@link RawTripData} instance coming from the `SaveTripModal` component.
+ * @param {!string[]} curAccepCollabUidArr Array stored in the field
+ *     `accepted_collaborator_uid_arr` of the {@link TripData} obj
+ *     `curTripData`.
+ * @return {!string[]} Array containing the set intersection of two params
+ *     `formCollabUidArr` and `curAccepCollabUidArr` converted to sets.
+ */
+export function getCurAcceptedCollabUidArr(formCollabUidArr, curAccepCollabUidArr) {
+  const formCollabUidSet = new Set(formCollabUidArr);
+  const curAccepCollabUidSet = new Set(curAccepCollabUidArr);
+  return [...formCollabUidSet].filter(el => curAccepCollabUidSet.has(el));
+}
+
+/**
+ * Returns the new `pending_collaborator_uid_arr` field value for the updated
+ * trip document.
+ *
+ * @param {!string[]} formCollabUidArr Array of the collaborator uids
+ *     corresponding to the emails in the `collaborator_email_arr` field of the
+ *     {@link RawTripData} instance coming from the `SaveTripModal` component.
+ * @param {!string[]} curAccepCollabUidArr Array stored in the field
+ *     `accepted_collaborator_uid_arr` of the {@link TripData} obj
+ *     `curTripData`.
+ * @return {!string[]} Array containing the set difference of two params
+ *     `formCollabUidArr` and `curAccepCollabUidArr` converted to sets.
+ */
+export function getCurPendingCollabUidArr(formCollabUidArr, curAccepCollabUidArr) {
+  const formCollabUidSet = new Set(formCollabUidArr);
+  const curAccepCollabUidSet = new Set(curAccepCollabUidArr);
+  return [...formCollabUidSet].filter(el => !curAccepCollabUidSet.has(el));
+}
```

Comment:
Accep -> Accept?

Reply:
I went with Accepted for all instances since that is the name of the field in the trip document.